### PR TITLE
Add `tls_server_name` provider options.

### PR DESCRIPTION
**Pull request by bartoszj**
### Description

Add support for `tls_server_name`.

Sometimes it is needed to connect to Kubernetes cluster using address which is not included in certificates SNIs. like IP addresses, SSH tunnels, etc. `insecure` parameter can be used in such scenarios, but it disable certificate validation. By using `tls_server_name` it is still possible to validate server certificate.

### Acceptance tests
- [X] Have you added an acceptance test for the functionality being added?
- [X] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note
Add `tls_server_name` provider options.
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: kubernetes/provider.go [hunk 1]
```diff
@@ -54,11 +54,11 @@ func Provider() *schema.Provider {
 				DefaultFunc: schema.EnvDefaultFunc("KUBE_INSECURE", false),
 				Description: "Whether server should be accessed without verifying the TLS certificate.",
 			},
-			"client_certificate": {
```
Comment (arybolovlev) [inline kubernetes/provider.go:57]: It looks like by accident you have overwritten the `client_certificate` option with a new one.

**Comment by arybolovlev** [comment]
Hi @bartoszj,

Thank you for your contribution. I have left one comment there. Could you please look at it?

Thank you!

**Comment by bartoszj** [comment]
> Hi @bartoszj,
> 
> Thank you for your contribution. I have left one comment there. Could you please look at it?
> 
> Thank you!

Hello. Unfortunately I cannot see your comment :( Can you add it once again?

File: kubernetes/provider.go (hunk 1 above)
Comment (bartoszj) [reply kubernetes/provider.go:57]: Fixed. Should be fine now

File: manifest/provider/provider_config.go [hunk 2]
```diff
@@ -54,6 +54,17 @@ func GetProviderConfigSchema() *tfprotov5.Schema {
 				DescriptionKind: 0,
 				Deprecated:      false,
 			},
+			{
+				Name:            "tls_server_name",
+				Type:            tftypes.String,
+				Description:     "Passed to the server for SNI and is used in the client to check server certificates against.",
```
Comment (jrhouston) [inline manifest/provider/provider_config.go:60]: This needs to match exactly what's in the schema for the main provider above, so this needs to be:

```
Server name passed to the server for SNI and is used in the client to check server certificates against.
```

It's not super intuitive, but when you run the provider in debug mode you can see the failure in the logs:

```
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64: 2022/03/21 12:07:35 got a different provider schema from two servers (*schema.GRPCProviderServer, *provider.RawProviderServer). Provider schemas must be identical across providers. Diff:   &tfprotov5.Schema{
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   	Version: 0,
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   	Block: &tfprotov5.SchemaBlock{
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		Version: 0,
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		Attributes: []*tfprotov5.SchemaAttribute(Inverse(cmpopts.SortSlices, []*tfprotov5.SchemaAttribute{
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		... // 10 identical elements
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		&{Name: "password", Type: s"tftypes.String", Description: "The password to use for HTTP basic authentication when accessing"..., Optional: true, ...},
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		&{Name: "proxy_url", Type: s"tftypes.String", Description: "URL to the proxy to be used for all API requests", Optional: true, ...},
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		&{
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		Name: "tls_server_name",
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		Type: s"tftypes.String",
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		Description: strings.Join({
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64: - 		"P",
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64: + 		"Server name p",
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		"assed to the server for SNI and is used in the client to check s",
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		"erver certificates against.",
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		}, ""),
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		Required: false,
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		Optional: true,
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		... // 4 identical fields
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		},
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		&{Name: "token", Type: s"tftypes.String", Description: "Token to authenticate an service account", Optional: true, ...},
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		&{Name: "username", Type: s"tftypes.String", Description: "The username to use for HTTP basic authentication when accessing"..., Optional: true, ...},
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		})),
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		BlockTypes:  {&{TypeName: "exec", Block: &{Attributes: Inverse(cmpopts.SortSlices, []*tfprotov5.SchemaAttribute{&{Name: "api_version", Type: s"tftypes.String", Required: true}, &{Name: "args", Type: s"tftypes.List[tftypes.String]", Optional: true}, &{Name: "command", Type: s"tftypes.String", Required: true}, &{Name: "env", Type: s"tftypes.Map[tftypes.String]", Optional: true}})}, Nesting: s"LIST", MaxItems: 1}, &{TypeName: "experiments", Block: &{Attributes: {&{Name: "manifest_resource", Type: s"tftypes.Bool", Description: "Enable the `kubernetes_manifest` resource.", Optional: true, ...}}, Description: "Enable and disable experimental features."}, Nesting: s"LIST", MaxItems: 1}},
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		Description: "",
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   		... // 2 identical fields
2022-03-21T12:07:35.594-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   	},
2022-03-21T12:07:35.595-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:   }
2022-03-21T12:07:35.595-0400 [DEBUG] provider.terraform-provider-kubernetes_9.9.9_darwin_amd64:
```

**Review by jrhouston — CHANGES_REQUESTED** [review]

**Review by BBBmau — APPROVED** [review]
Thank you for contributing!

**Review by jrhouston — APPROVED** [review]

**Pull request merged**